src/main/java/eu/europeana/metis/sandbox/service/workflow/InternalValidationServiceImpl.java

```diff
@@ -2,35 +2,63 @@
 import eu.europeana.validation.service.ValidationExecutionService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 import org.springframework.stereotype.Service;
 
+import java.nio.charset.StandardCharsets;
+import java.time.LocalDateTime;
+import java.util.Optional;
+
 @Service
 class InternalValidationServiceImpl implements InternalValidationService {
 
-  private static final String SCHEMA = "EDM-INTERNAL";
+    private static final Logger LOGGER = LoggerFactory.getLogger(InternalValidationServiceImpl.class);
+    private static final String SCHEMA = "EDM-INTERNAL";
 
-  private final ValidationExecutionService validator;
+    private final ValidationExecutionService validator;
+    private final PatternAnalysisService<Step> patternAnalysisService;
+    private final ExecutionPointService executionPointService;
 
-  public InternalValidationServiceImpl(
-      ValidationExecutionService validator) {
-    this.validator = validator;
-  }
+    public InternalValidationServiceImpl(
+            ValidationExecutionService validator, PatternAnalysisService<Step> patternAnalysisService,
+            ExecutionPointService executionPointService) {
+        this.validator = validator;
+        this.patternAnalysisService = patternAnalysisService;
+        this.executionPointService = executionPointService;
+    }
 
-  @Override
-  public RecordInfo validate(Record record) {
-    requireNonNull(record, "Record must not be null");
+    @Override
+    public RecordInfo validate(Record recordToValidate) {
+        requireNonNull(recordToValidate, "Record must not be null");
 
-    var content = record.getContentInputStream();
-    var validationResult = validator.singleValidation(SCHEMA, null, null, content);
-    if (!validationResult.isSuccess()) {
-      throw new RecordValidationException(validationResult.getMessage(),
-          validationResult.getRecordId(), validationResult.getNodeId());
+        var content = recordToValidate.getContentInputStream();
+        var validationResult = validator.singleValidation(SCHEMA, null, null, content);
+        if (!validationResult.isSuccess()) {
+            throw new RecordValidationException(validationResult.getMessage(),
+                    validationResult.getRecordId(), validationResult.getNodeId());
+        }
+        try {
+            generateAnalysis(recordToValidate.getDatasetId(), recordToValidate.getContent());
+        } catch (PatternAnalysisException e) {
+            LOGGER.error("An error occurred while processing pattern analysis with record id {}", recordToValidate.getEuropeanaId());
+        }
+        return new RecordInfo(recordToValidate);
     }
 
-    return new RecordInfo(record);
-  }
+    private void generateAnalysis(String datasetId, byte[] recordContent) throws PatternAnalysisException {
+        Optional<ExecutionPoint> firstOccurrence = executionPointService.getExecutionPoint(datasetId, Step.VALIDATE_INTERNAL.toString());
```

Comment:
We checked together with Joana this method. we will continue to solve the sync with MET-4470